core/src/main/java/io/grpc/StreamTracer.java

```diff
@@ -62,32 +62,45 @@ public void inboundMessage() {
   }
 
   /**
+   * An outbound message has been completely serialized, revealing its wire size.
+   */
+  public void outboundMessageSerialized(long wireSize) {
```

Comment:
We can't support this accurately with stream-wide compression, because we won't want to flush the compressor after every message. So we'll just know "the last three messages are X bytes on the wire."